include/ccf/crypto/key_pair.h

```diff
@@ -60,7 +61,8 @@ namespace crypto
       const Pem& signing_request,
       const std::string& valid_from,
       const std::string& valid_to,
-      bool ca = false) const = 0;
+      bool ca = false,
+      bool signed_by_issuer = false) const = 0;
```

Comment:
`Signer signed_by = Signer::SUBJECT/Signer::ISSUER` may be clearer and potentially more extensible?

Reply:
Why would the issuer ever sign the CSR?

Reply:
The issuer signs the CSR on behalf of the previous service, which isn't available anymore. It's a somewhat usual thing to do, but in our case we could get around it by immediately issuing a CSR in the recovery transaction, so that a future service identity can then sign it after the next recovery. I don't think that's necessary, but it's definitely an option, especially if we do not support old ledgers.